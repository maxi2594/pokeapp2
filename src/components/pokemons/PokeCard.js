import React from 'react'
import { Link } from 'react-router-dom';
import { useFetchPokes } from '../../hooks/useFetchPokes';

export const PokeCard = ({ pokemon, sprites }) => {
    
    const styles = {
        cardCenter: {
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'center',
            backgroundColor: 'rgb(63,67,71)',
            borderRadius: '100px'
        },
        cardStyles: {
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'center',
            border: '2px solid grey',
            backgroundColor: '#FAFAFA',
            maxWidth: '400px',
            margin: '5px',
            padding: '10px'
        },
        cardImg: {
            width: '150px',
            height: '150px',
        }
    }

    return (
        
        <div className="card col-sm-6 mb-2 animate__animated animate__fadeIn animate__delay-.5s" style={ styles.cardStyles} >
            <div className="col-md-6">
                <div style={styles.cardCenter}>

                    <img
                        style={ styles.cardImg}
                        src={ sprites.front_default } 
                        className="card-img"
                        alt={ pokemon.name }/>

                </div>
            </div>
            <div className="col-md-8">
                <div className="card-body">

                    <h2
                        style={{textAlign: 'center'}}
                    > { pokemon.name } 
                    </h2>

                    <hr></hr>    
                </div>
            </div>
        </div>
)
}

