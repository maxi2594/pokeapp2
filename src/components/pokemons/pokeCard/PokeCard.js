import React from 'react'
import { Link } from 'react-router-dom';
import PropTypes from "prop-types";

import './pokeCard.css'

export const PokeCard = ({ pokemon, sprites, id }) => {


    return (

        <div className="card col-sm-6 mb-2 cardStyles animate__animated animate__fadeIn animate__delay-.5s">
            <div className="col-md-6">
                <div className='cardCenter'>

                    <Link to={`/pokes/character/${id}`}>
                        <img
                            className='cardImg card-img'
                            src={sprites.front_default}
                            alt={pokemon.name}
                        />
                    </Link>
                </div>
            </div>
            <div className="col-md-8">
                <div className="card-body">

                    <h2
                        style={{ textAlign: 'center' }}
                    > {pokemon.name}
                    </h2>

                    <hr></hr>
                </div>
            </div>
        </div>

    )
}

PokeCard.propTypes = {
    pokemon: PropTypes.object.isRequired,
    sprites: PropTypes.object.isRequired,
    id: PropTypes.number.isRequired
}
