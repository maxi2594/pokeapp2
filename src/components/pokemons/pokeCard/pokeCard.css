.cardCenter {
  align-items: center;
  display: flex;
  justify-content: center;
  background-color: rgb(63, 67, 71);
  border-radius: 100px;
}
.cardStyles {
  align-items: center;
  display: flex;
  justify-content: center;
  border: 2px solid grey;
  background-color: #fafafa;
  max-width: 400px;
  margin: 5px;
  padding: 10px;
}
@media (max-width: 900px) {
  .cardStyles {
    width: 80%;
  }
}
.cardImg {
  width: "150px";
  height: "150px";
}
