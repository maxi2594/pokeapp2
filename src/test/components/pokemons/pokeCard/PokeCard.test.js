import { shallow } from 'enzyme'
import React from 'react'
import { PokeCard } from '../../../../components/pokemons/pokeCard/PokeCard'


describe('Test in <PokeCard/>', () => {

    const values = {
        pokemon: { name: 'charizard' },
        sprites: { front_default: 'https://fakeUrl' },
        id: 45
    }

    const wrapper = shallow(<PokeCard {...values} />)

    test('should shown properly', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('should have an img tag(html) and src / alt props', () => {

        const props = wrapper.find('img').props();

        expect(wrapper.find('img').exists()).toBe(true);

        expect(props.src).toBe(values.sprites.front_default);
        expect(props.alt).toBe(values.pokemon.name);
        expect(wrapper.find('h2').text().trim()).toBe(values.pokemon.name);

    })

})
