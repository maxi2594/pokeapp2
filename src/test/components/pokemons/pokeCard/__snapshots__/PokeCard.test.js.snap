// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <PokeCard/> should shown properly 1`] = `
<div
  className="card col-sm-6 mb-2 cardStyles animate__animated animate__fadeIn animate__delay-.5s"
>
  <div
    className="col-md-6"
  >
    <div
      className="cardCenter"
    >
      <Link
        to="/pokes/character/45"
      >
        <img
          alt="charizard"
          className="cardImg card-img"
          src="https://fakeUrl"
        />
      </Link>
    </div>
  </div>
  <div
    className="col-md-8"
  >
    <div
      className="card-body"
    >
      <h2
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
         
        charizard
      </h2>
      <hr />
    </div>
  </div>
</div>
`;
